api/objects.proto

```diff
@@ -222,3 +222,35 @@ message Cluster {
 	// a new key is allocated on key rotation.
 	uint64 encryption_key_lamport_clock = 6;
 }
+
+// Secret represents a secret that should be passed to a container or a node,
+// and is immutable. It wraps the `spec` provided by the user with useful
+// information that is generated from the secret data in the `spec`, such as
+// the digest and size of the secret data.
+message Secret {
```

Comment:
Makes sense.  Is it possible for two secrets to have the same version ID, if they came in at the same time?  (e.g. both did the calculation)  Would we then tie-break using creation time (which seems ok, since if there was a to add secrets one has to win anyway, and the next secret will have a higher value).